sdk/python/feast/infra/offline_stores/contrib/mssql_offline_store/tests/data_source.py

```diff
@@ -0,0 +1,117 @@
+from typing import Dict, List
+
+import pandas as pd
+import pytest
+from sqlalchemy import create_engine
+from testcontainers.core.container import DockerContainer
+from testcontainers.core.waiting_utils import wait_for_logs
+from testcontainers.mssql import SqlServerContainer
+
+from feast.data_source import DataSource
+from feast.infra.offline_stores.contrib.mssql_offline_store.mssql import (
+    MsSqlServerOfflineStoreConfig,
+    _df_to_create_table_sql,
+)
+from feast.infra.offline_stores.contrib.mssql_offline_store.mssqlserver_source import (
+    MsSqlServerSource,
+)
+from feast.saved_dataset import SavedDatasetStorage
+from tests.integration.feature_repos.universal.data_source_creator import (
+    DataSourceCreator,
+)
+
+MSSQL_USER = "SA"
+MSSQL_PASSWORD = "yourStrong(!)Password"
+
+
+# This is the sql container to use if your machine doesn't support the official msql docker container.
+@pytest.fixture(scope="session")
```

Comment:
I kept it in there because it might be useful for mac m1 developers. I did all of my develop on gitpod because the SqlServerContainer image doesn't work with m1...

Reply:
you should be able to use SqlServerContainer by passing in an image override (`SqlServerContainer(image="mcr.microsoft.com/azure-sql-edge:1.0.6"...)`)

Would also prefer using the fixture here instead of what you do of creating a new container

Reply:
in which case you'll need to add back that import in the __init__.py in this directory

Reply:
added back in